File: gc-app/src/main/java/rocks/frick/gc/application/UpdateActivityNameService.java
package rocks.frick.gc.application;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import rocks.frick.gc.application.spi.ActivityRepository;

public class UpdateActivityNameService {

  private static final Logger log = LoggerFactory.getLogger(UpdateActivityNameService.class);

  private final ActivityRepository activityRepository;

  public UpdateActivityNameService(ActivityRepository activityRepository) {
    this.activityRepository = activityRepository;
  }

  public void updateActivityName(String activityId, String newName) {
    log.debug("Updating name of activity with ID {} to '{}'", activityId, newName);
    activityRepository.updateActivityName(activityId, newName);
  }
}

File: gc-core/src/main/scala/gc/application/UpdateActivityNameService.scala
package gc.application

import gc.spi.UpdateActivityName
import com.typesafe.scalalogging.Logger

class UpdateActivityNameService(repo: UpdateActivityName) {
  private val log = Logger("UpdateActivityNameService")

  def updateActivityName(id: String, name: String): Unit = {
    log.debug(s"[CORE] Updating activity name for ID: $id to '$name'")
    repo.updateActivityName(id, name)
  }
}
